### PR TITLE
MethodOrderer.Random seed generation and handling improvements

### PR DESCRIPTION
**Pull request by aepfli**
## Overview

As discussed in #1805 it is hard to rerun exactly the same test if the MethodOrderer.Random has been used. 

1. It used to generate the fallback seed per test execution and based on `System.nanoTime()`, which made it "random" through out the run.
2. It did not provide any log output to the user, to retrieve the information what seed he should use.

In this pullrequest, we change the initialization of the seed to a static one, so it will be the same seed for all tests. Additionally we log the generated seed at the beginning with an log level `CONFIG`.

During the implementation we also found an issue regarding concurrent runs with random seeds, and that we can not guarantee the order based on thread assignment. Hence that the team decided that if a test is ran with MethodOrderer.Random it will be always executed in the same thread.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/OrderedMethodTests.java [hunk 1]
```diff
@@ -123,34 +124,61 @@ private void assertOrderAnnotationSupport(Class<?> testClass) {
...
+		assertThat(uniqueSequences).size().isEqualTo(1);
 	}
 
 	@Test
 	@TrackLogRecords
-	void randomWithBogusSeed(LogRecordListener listener) {
-		String seed = "explode";
-		String expectedMessage = "Failed to convert configuration parameter [" + Random.RANDOM_SEED_PROPERTY_NAME
-				+ "] with value [" + seed + "] to a long. Using System.nanoTime() as fallback.";
+	@Disabled("needs to be loaded within own Classloader, not sure how to verify")
```
Comment (aepfli) [inline junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/OrderedMethodTests.java]: i am not sure, how i can properly test this. i am not sure how to use a custom class loader in combination with @trackLogRecords. Guidance is highly appreciated

Comment (sbrannen) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/OrderedMethodTests.java]: If you cannot get the testing of the logging for the static initialization to work, don't worry about it. We can always add a test for that later after merging the PR.

Comment (aepfli) [reply junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/OrderedMethodTests.java]: :+1:

**Comment by aepfli** [comment]
i just took a look at @marcphilipp presentation, and i figured out that some information on my side needs to be adapted, like:
- the information regarding same thread and methodOrderer.random 
- checking if i really need the override of the executionMode method. or maybe change it to return Same_Thread always, as all 3 default orderer are using Same_Thread now
- adapt documentation

**Comment by aepfli** [comment]
i think i am fine now :D review is highly appreciated

maybe i should also squash all my commits, i made quiet a mess

**Comment by sbrannen** [comment]
> i think i am fine now :D review is highly appreciated

Great. I'll take a look.

> maybe i should also squash all my commits, i made quiet a mess

Don't worry about it. We will squash and rebase when we merge.

**Comment by sbrannen** [comment]
This has been merged into `master`.

Thanks!

**Comment by aepfli** [comment]
thank you!

**Pull request merged**